### PR TITLE
Internal Review: PortChannel support for REST and subscription 

### PR DESCRIPTION
**Pull request by Satoru-Shinohara**
Support for PortChannel at REST and subscription level along with test automation. 

New file: translib/transformer/sw_portchannel.go
New Test File: translib/transformer/portchannel_openconfig_test.go

TestApp Result: (Part of IPv4 and v6 tests fail locally as they are made to pass azure-pipeline tests which does not recognize STATE_DB)
[testapp_pc.log](https://github.com/user-attachments/files/15828014/testapp_pc.log)

Subscription Logs:
(At interface[name=PortChannel###]/config and interface[name=PortChannel###]/openconfig-if-aggregate level)

[on_change_config.log](https://github.com/user-attachments/files/15828040/on_change_config.log)
[sample_config.log](https://github.com/user-attachments/files/15828041/sample_config.log)
[target_defined_config.log](https://github.com/user-attachments/files/15828042/target_defined_config.log)

[on_change_pc.log](https://github.com/user-attachments/files/15828044/on_change_pc.log)
[sample_pc.log](https://github.com/user-attachments/files/15828045/sample_pc.log)
[target_defined_pc.log](https://github.com/user-attachments/files/15828046/target_defined_pc.log)

### CODE REVIEW

**Comment by Satoru-Shinohara** [comment]
The large comment block at translib/transformer/portchannel_openconfig_test.go will be deleted before merge. 
Missed pre-PR and would rather delete along with any other changes brought by review.

File: translib/transformer/portchannel_openconfig_test.go [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+	"testing"
+	"time"
+)
+
+func Test_openconfig_portchannel(t *testing.T) {
+	var url, url_input_body_json string
+
+	t.Log("\n\n+++++++++++++ CONFIGURING PORTCHANNEL ++++++++++++")
+
+	t.Log("\n\n--- PUT to Create PortChannel 111 ---")
```
Comment (kwangsuk) [inline translib/transformer/portchannel_openconfig_test.go:35]: Please add the subsequent GET test to ensure the expected results read after PUT/PATCH/DELETE/POST.

Comment (Satoru-Shinohara) [reply translib/transformer/portchannel_openconfig_test.go:35]: Thank you! Addressed by adding test after min-links delete and also after portchannel creation to make sure creation succeeded. 
Please let me know if there is anything else to change!

File: models/yang/extensions/openconfig-interfaces-deviation.yang [hunk 2]
```diff
@@ -255,6 +247,30 @@ module openconfig-interfaces-deviation {
...
 
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-eth:ethernet/oc-vlan:switched-vlan {
+    deviate not-supported;
+  }
+  
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-lag:aggregation/oc-vlan:switched-vlan {
+    deviate not-supported;
+  }
+
+  deviation /oc-intf:interfaces/oc-intf:interface/oc-lag:aggregation/oc-lag:config/oc-lag:lag-type {
```
[the remark this answers is not in the document]
Comment (Satoru-Shinohara) [reply models/yang/extensions/openconfig-interfaces-deviation.yang:258]: The lag-type is marked as not supported due to the following reason: 
In the scoping stage of the project, we created a portchannel and added an Ethernet member via CLICK commands. 
Dumping CONFIG_DB afterwards resulted in this output with no lag-type (please see block below). We believe all portchannel is automatically handled as static in community. 
the lag-type attribute is also not present in community version of [sonic-portchannel.yang](https://github.com/sonic-net/sonic-mgmt-common/blob/81bf799d8e346817ccd335dcd4da6258f8edd7f1/cvl/testdata/schema/sonic-portchannel.yang#L27) under the PORTCHANNEL container. 


```
   "PORTCHANNEL_MEMBER|PortChannel200|Ethernet200": {
    "expireat": 1710883577.2006261,
    "ttl": -0.001,
    "type": "hash",
    "value": {
      "NULL": "NULL"
    }
  },
  "PORTCHANNEL_MEMBER|PortChannel200|Ethernet204": {
    "expireat": 1710883577.2003803,
    "ttl": -0.001,
    "type": "hash",
    "value": {
      "NULL": "NULL"
    }
  },
  "PORTCHANNEL|PortChannel200": {
    "expireat": 1710883577.2014394,
    "ttl": -0.001,
    "type": "hash",
    "value": {
      "admin_status": "up",
      "fast_rate": "false",
      "lacp_key": "auto",
      "min_links": "1",
      "mtu": "9100"
    }
  },
```

[the remark this answers is not in the document]
Comment (kwangsuk) [reply models/yang/extensions/openconfig-interfaces-deviation.yang:258]: Okay, thanks for update.

**Review by kwangsuk — APPROVED** [review]

**Pull request merged**